### PR TITLE
Bromjahm event & achievement Soul Power

### PR DESCRIPTION
**Pull request by ghost**
Based on PR #13606 (without SQL)

### CODE REVIEW

**Comment by tkrokli** [comment]
This replaces PR #13606 due to the OP abandoning the original PR.

**Comment by tkrokli** [comment]
I managed to make a correction on the NPC name in the commit title, but I guess the NPC name typo in the PR name can't be changed that easily. Or do you admins have powers to fix typos in the PR title? :)

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 1]
```diff
@@ -176,6 +201,8 @@ class boss_bronjahm : public CreatureScript
 
                 DoMeleeAttackIfReady();
             }
+        private:
+            uint32 _soulFragmentsSpawned;
```
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: > **jackpoz** added a note on [8 Dec 2014](https://github.com/TrinityCore/TrinityCore/pull/13606/files#r21454627)
please initialize all fields in the constructor

Sorry, I don't know about constructors. Is that part of the definitions in the header?

Comment (NorthStrider) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: he want you to unitialize this field at  https://github.com/tkrokli/TrinityCore/blob/Bromjahm-Soul-Power/src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp#L74

Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: :+1: OK, thanks for the clarification. It is about setting the `uint32 _soulFragmentsSpawned` to zero or some base value, right? So now I just have to Google "how to initialize all fields in the constructor uint32 ". Got any favorite pages about initializing the constructor? :memo: :notebook:

Comment (MitchesD) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: add _soulFragmentsSpawned = 0; to the constructor or use Initialize() method

check for example this: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/scripts/Northrend/DraktharonKeep/boss_trollgore.cpp#L81

Comment (NorthStrider) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: no. initialize the variable _soulFragmentsSpawned = 0; withing the brackets and thats it :)

**Comment by tkrokli** [comment]
> no. initialize the variable _soulFragmentsSpawned = 0; within the brackets and thats it :) so
```c++
boss_bronjahmAI(Creature* creature) : BossAI(creature, DATA_BRONJAHM)
{
DoCast(me, SPELL_SOULSTORM_CHANNEL, true);
_soulFragmentsSpawned = 0;
}
```

Thanks a bunch, dude. It feels awkward to be clueless when it is a piece of cake to you.
Have some cake. :cake: =^.^=

**Pull request closed without merging**